### PR TITLE
Feat/set the bot for listening when the poll is ended

### PR DESCRIPTION
**Pull request by YoelFerreyra**

### CODE REVIEW

File: src/events/qaMention.js [hunk 1]
```diff
@@ -1,9 +1,13 @@
 const { Events } = require('discord.js');
 const { DISCORD_CONFIG } = require('../config');
+const { pollResults } = require('../utils/poll-vote');
 
 module.exports = {
   name: Events.MessageCreate,
   async execute(message) {
+
+    if (message.embeds[0]?.data?.type === 'poll_result') await pollResults(message)
```
Comment (heliomar-pena) [inline src/events/qaMention.js:9]: Idk if it is possible to add here an extra validation like if author of message was BOT and if embed includes the ID of the author of the PR. 

Since if it is not the BOT the author, then we shouldn't assume it is a points votation

And if it doesn't have the ID of the author of the PR, we will don't know what is the user that we should give the points

File: src/events/qaMention.js [hunk 2]
```diff
@@ -1,9 +1,13 @@
 const { Events } = require('discord.js');
 const { DISCORD_CONFIG } = require('../config');
+const { pollResults } = require('../utils/poll-vote');
 
 module.exports = {
   name: Events.MessageCreate,
   async execute(message) {
+
+    if (message.embeds[0]?.data?.type === 'poll_result') await pollResults(message)  
+
```
Comment (heliomar-pena) [inline src/events/qaMention.js:10]: Move this code to another event. Create a new file in events folder called pollVotationResults.js or something similar, then export a new event like this but only for poll results.

File: src/events/qaMention.js [hunk 3]
```diff
@@ -1,9 +1,13 @@
 const { Events } = require('discord.js');
 const { DISCORD_CONFIG } = require('../config');
+const { pollResults } = require('../utils/poll-vote');
 
 module.exports = {
   name: Events.MessageCreate,
```
Comment (heliomar-pena) [inline src/events/qaMention.js:6]: Maybe we could explore instead of using MessageCreate, using MessageUpdate. Idk if poll message is updated when it ended, so we probably could access to more properties and also we could possibly get the result even in DRAWS.

This could be handle on another PR since this PR is all based on messageCreate event and the embed that appears when poll end

File: src/utils/poll-vote.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+const { tagIds } = require("../config");
+
+
```
Comment (heliomar-pena) [inline src/utils/poll-vote.js:3]: Remove extra space

```suggestion
```

**Review by heliomar-pena — CHANGES_REQUESTED** [review]
Good work with the investigation! And thanks for the refactor, moving all tagsIds to an object.

I leave some comments to improve the code

File: src/events/qaMention.js (hunk 1 above)
Comment (joset98) [reply src/events/qaMention.js:9]: Maybe something like that
>image

File: src/events/qaMention.js (hunk 3 above)
Comment (heliomar-pena) [reply src/events/qaMention.js:6]: I created the task https://github.com/TeamNovaSoft/discord-bot/issues/30

File: src/commands/utility/vote-points.js [hunk 5]
```diff
@@ -1,23 +1,33 @@
...
+          text: `How much cost this task? | ${user.id}`,
           emoji: { name: '🧮' },
         },
         allowMultiselect: false,
-        duration: 24,
+        duration: 1,
         answers: VOTE_POINTS_ANSWERS,
       },
-    });
+    });
```
Comment (heliomar-pena) [inline src/commands/utility/vote-points.js:31]:
```suggestion
    });
```

File: src/events/pollVotationResults.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+module.exports = {
+  name: Events.MessageCreate,
+  async execute(message) {
+
+    if (message.author.bot) {
+      return;
+    }
+
+    if (message.embeds[0]?.data?.type === 'poll_result') await pollResults(message)
```
Comment (heliomar-pena) [inline src/events/pollVotationResults.js:12]: Maybe we can validate all we need on first conditional
 
```suggestion
    if (message.author.bot && message.embeds[0]?.data?.type !== 'poll_result') {
      return;
    }

    await pollResults(message)
```

File: src/events/qaMention.js (hunk 2 above)
Comment (heliomar-pena) [inline src/events/qaMention.js:10]: this can be removed now, since you already created a [pollVotationResults](https://github.com/TeamNovaSoft/discord-bot/pull/29/files#diff-e7608383e0544452f227a7ad2e8dbeb02f7fa564b31c80cfe2138db3256a892aR4-R12) events

File: src/utils/poll-vote.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    } else {
+      await channel.send(`The draw is not suported`);
+    }
+
+  } else {
+    return console.error('Channel not found:', message.channelId);
+  }
+}
+
+module.exports = { pollResults }
```
Comment (heliomar-pena) [inline src/utils/poll-vote.js:31]: This seem like it is not reusable?, maybe you can move it directly to https://github.com/TeamNovaSoft/discord-bot/pull/29/files#diff-e7608383e0544452f227a7ad2e8dbeb02f7fa564b31c80cfe2138db3256a892aR4-R12

File: src/commands/utility/vote-points.js [hunk 8]
```diff
@@ -1,23 +1,33 @@
...
+    interaction.followUp({
       poll: {
         question: {
-          text: 'How much cost this task?',
+          text: `How much cost this task? | ${user.id}`,
           emoji: { name: '🧮' },
         },
         allowMultiselect: false,
-        duration: 24,
+        duration: 1,
```
[the remark this answers is not in the document]
Comment (YoelFerreyra) [reply src/commands/utility/vote-points.js:28]: Complete

File: src/commands/utility/vote-points.js (hunk 5 above)
Comment (YoelFerreyra) [reply src/commands/utility/vote-points.js:31]: Complete

File: src/events/pollVotationResults.js (hunk 6 above)
Comment (YoelFerreyra) [reply src/events/pollVotationResults.js:12]: Complete

File: src/events/qaMention.js (hunk 2 above)
Comment (YoelFerreyra) [reply src/events/qaMention.js:10]: Complete

**Review by joset98 — DISMISSED** [review]

**Review by heliomar-pena — CHANGES_REQUESTED** [review]
just some small changes

File: src/events/qaMention.js [hunk 9]
```diff
@@ -4,6 +4,10 @@ const { DISCORD_CONFIG } = require('../config');
...
   async execute(message) {
+    if (!message.embeds[0]?.data?.type === 'poll_result') {
+      return;
+    }
+
```
[the remark this answers is not in the document]
Comment (YoelFerreyra) [reply src/events/qaMention.js:10]: Complete

File: src/events/pollVotationResults.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+  async execute(message) {
+    const { embeds, author, client, channelId } = message;
+
+    if (author.bot) {
+      return;
+    }
+
+    if (author.bot && embeds[0]?.data?.type !== 'poll_result') {
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (YoelFerreyra) [reply src/events/pollVotationResults.js:15]: Complete

**Review by heliomar-pena — APPROVED** [review]
Good work!

File: src/events/pollVotationResults.js [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+    );
+    const parts = userMentionField?.value.split('|').map((part) => part.trim());
+    const userMentioned = `<@${parts[1]}>`;
+
+    const channel = await client.channels.fetch(channelId);
+
+    if (channel) {
+      if (embeds[0].fields.length > 3) {
+        const finalResult = embeds[0].fields.find(
+          (result) => result.name === 'victor_answer_text'
```
Comment (Israel-Laguan) [inline src/events/pollVotationResults.js:24]: Please check this condition, it mention "victor" I think is not the desired condition

File: src/events/pollVotationResults.js [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+              `<@&${tagIds.boostedPointTagId}> ${userMentioned}`
+            );
+          })
+        );
+      } else {
+        await channel.send(`The draw is not suported`);
+      }
+    } else {
+      return console.error('Channel not found:', channelId);
+    }
```
Comment (Israel-Laguan) [inline src/events/pollVotationResults.js:39]: Please send the error to the user too because it can fail silently

**Review by Israel-Laguan — APPROVED** [review]

**Review by heliomar-pena — APPROVED** [review]
LGTM. good work!
